python/paddle/amp/grad_scaler.py

```diff
@@ -11,22 +11,43 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+from __future__ import annotations
 
 import warnings
 from collections import defaultdict
 from enum import Enum
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    TypedDict,
+)
 
 import numpy as np
 
 import paddle
 from paddle import _C_ops, _legacy_C_ops
 from paddle.base import core, unique_name
 from paddle.base.data_feeder import check_type
-from paddle.base.framework import _dygraph_tracer, in_pir_mode
+from paddle.base.framework import Operator, _dygraph_tracer, in_pir_mode
 from paddle.framework import in_dynamic_mode
 
 from .auto_cast import amp_global_state
 
+if TYPE_CHECKING:
+    from paddle import Tensor
+    from python.paddle.optimizer.optimizer import Optimizer
+
+
+class _ScaleStateDict(TypedDict):
+    scale: Tensor
+    incr_ratio: float
+    decr_ratio: float
+    incr_every_n_steps: int
+    decr_every_n_nan_or_inf: int
+    incr_count: int
+    decr_count: int
+    use_dynamic_loss_scaling: bool
```

Comment:
```suggestion
    class _ScaleStateDict(TypedDict):
        scale: Tensor
        incr_ratio: float
        decr_ratio: float
        incr_every_n_steps: int
        decr_every_n_nan_or_inf: int
        incr_count: int
        decr_count: int
        use_dynamic_loss_scaling: bool
```

放到 `TYPE_CHECKING` 里面吧 ～

Reply:
收到